server/src/main/java/org/apache/iotdb/db/metadata/mnode/InternalMNode.java

```diff
@@ -169,6 +169,12 @@ public synchronized void replaceChild(String oldChildName, IMNode newChildNode)
       newChildNode.getAsEntityMNode().setUseTemplate(oldChildNode.isUseTemplate());
     }
 
+    if (newChildNode.isStorageGroup() && oldChildNode.isStorageGroup()) {
```

Comment:
It is better to make these logic as part of the implementation of subclass.

Otherwise, we need to add more `if-else` here if we add new type of IMNode, which is easy to be missed...

Reply:
Fixed. I added a new interface, moveDataToNewNode(IMNode newMNode), to IMNode and the replace method will invoke the oldChildNode.moveDataToNewNode(newChildNode). The moveDataToNewNode will be overrided by different types of mnode.